src/sentry/eventstream/kafka/postprocessworker.py

```diff
@@ -0,0 +1,134 @@
+    skip_consume: bool = False,
+) -> None:
+    if skip_consume:
+        logger.info("post_process.skip.raw_event", extra={"event_id": event_id})
+    else:
+        cache_key = cache_key_for_event({"project": project_id, "event_id": event_id})
+
+        post_process_group.delay(
+            is_new=is_new,
+            is_regression=is_regression,
+            is_new_group_environment=is_new_group_environment,
+            primary_hash=primary_hash,
+            cache_key=cache_key,
+            group_id=group_id,
+        )
+
+
+def _get_task_kwargs_and_dispatch(message: Message):
+    task_kwargs = _get_task_kwargs(message)
+    if not task_kwargs:
+        return None
+
+    _record_metrics(message.partition(), task_kwargs)
+    dispatch_post_process_group_task(**task_kwargs)
+
+
+class PostProcessForwarderWorker(AbstractBatchWorker):
+    """
+    Implementation of the AbstractBatchWorker which would be used for post process forwarder.
+    The current implementation creates a thread pool worker based on the concurrency parameter
+    because we want to be able to change the concurrency during runtime. This should be replaced
+    by a thread pool executor once stress tests experiments are over and we start using the
+    CLI arguments to set concurrency.
+    """
+
+    def __init__(self, concurrency: Optional[int] = 1) -> None:
+        self.__current_concurrency = concurrency
+        self.__executor = ThreadPoolExecutor(max_workers=self.__current_concurrency)
+
+    def process_message(self, message: Message) -> Optional[Future]:
+        """
+        Process the message received by the consumer and return the Future associated with the message. The future
+        is stored in the batch of batching_kafka_consumer and provided as an argument to flush_batch. If None is
+        returned, the batching_kafka_consumer will not add the return value to the batch.
+        """
+        return self.__executor.submit(_get_task_kwargs_and_dispatch, message)
+
+    def flush_batch(self, batch: Sequence[Future]) -> None:
+        """
+        For all work which was submitted to the thread pool executor, we need to ensure that if an exception was
+        raised, then we raise it in the main thread. This is needed so that processing can be stopped in such
+        cases.
+        """
+        for future in as_completed(batch):
+            exc = future.exception()
+            if exc is not None:
+                raise exc
+
+        # Check if the concurrency settings have changed. If yes, then shutdown the existing executor
+        # and create a new one with the new settings
+        new_concurrency = options.get(_CONCURRENCY_OPTION)
+        if new_concurrency != self.__current_concurrency:
+            logger.info(f"Switching post-process-forwarder to {new_concurrency} worker threads")
+            self.__executor.shutdown(wait=True)
+            self.__executor = ThreadPoolExecutor(max_workers=new_concurrency)
```

Comment:
Let's track a metric with the current threadPool size. the logger.info is not going to be visible anywhere. At least with a metric we will have visibility on the current state when we make changes.